### PR TITLE
Windows service shutdown fixes

### PR DESCRIPTION
**Pull request by arsing**
- Log error to event log *before* marking service as stopped.
  Otherwise the logs get swallowed.

- If `ModuleRuntime::stop` fails with `NotFound`, treat it as a success.
  Currently, if the user stops the service before it has started edge agent,
  the watchdog fails to stop the module runtime and returns an error.
  This causes the service to crash and restart instead of staying stopped.

  This is a global solution to a local problem, but since the watchdog lives in
  `edgelet_core` it can't check if `ModuleRuntime::stop()` failed with
  `edgelet_docker::ErrorKind::NotFound`

### CODE REVIEW

**Comment by arsing** [comment]
>This is a global solution to a local problem, but since the watchdog lives in
>`edgelet_core` it can't check if `ModuleRuntime::stop()` failed with
>`edgelet_docker::ErrorKind::NotFound`

Perhaps the cleaner solution would be to enforce `&<M as ModuleRuntime>::Error` is convertible to `edgelet_core::ModuleRuntimeErrorKind`, which would be something like `enum ModuleRuntimeErrorKind { NotFound, NotModified, ... , Other }`. It would solve this problem as well as the one fixed by #560

File: edgelet/iotedged/src/windows.rs [hunk 1]
```diff
@@ -87,13 +97,9 @@ fn run_as_service(_: Vec<OsString>) -> Result<(), Error> {
 
     // start running
     info!("Starting {} service.", IOTEDGED_SERVICE_NAME);
-    let result = main.run_until(shutdown_signal);
-
-    // let Windows know that we stopped
-    info!("Stopped {} service.", IOTEDGED_SERVICE_NAME);
-    update_service_state(status_handle, ServiceState::Stopped)?;
+    main.run_until(shutdown_signal)?;
```
[the remark this answers is not in the document]
Comment (arsing) [reply edgelet/iotedged/src/windows.rs]: The change to this file is the first bullet point. The second bullet point is the change to `runtime.rs`

**Comment by arsing** [comment]
Yes, I'm going to evaluate it today. If I like it I'll put it in this PR.

File: edgelet/iotedged/src/windows.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (damonbarry) [reply edgelet/iotedged/src/windows.rs]: I get it now, thanks.

**Comment by arsing** [comment]
Added a new enum in `edgelet_core` to represent a `NotFound` module runtime error, and reverted the change in `DockerModuleRuntime::stop`.

But, as discussed with Damon offline, it's not possible to remove the `downcast_ref::<DockerErrorKind>` added in #560 since `Fail` can only be downcast to `Sized` types. So I've left that code unchanged.

**Pull request merged**